AOAICustomSkills/function_app.py

```diff
@@ -0,0 +1,134 @@
+      response_values.append(api_response)
+    response_body = { "values": response_values }
+    response = func.HttpResponse(json.dumps(response_body, default=lambda obj: obj.__dict__))
+    response.headers['Content-Type'] = 'application/json'
+    return response
+
+# TODO: figure out how to add this into a different file later. It's currently causing interpreter errors when running locally.
+def call_chat_completion_model(request_body: dict, scenario: str):
+    SUMMARIZATION_HEADER = "summarization"
+    ENTITY_RECOGNITION_HEADER = "entity-recognition"
+    api_key = os.getenv("AZURE_INFERENCE_CREDENTIAL")
+    logging.info(f'the api key is: {api_key}')
+    headers = {
+        "Content-Type": "application/json",
+        "api-key": api_key,
+    }
+    # default our chat completion context to be for summarization
+    chat_completion_system_context = {}
+    messages = []
+    if scenario == SUMMARIZATION_HEADER:
+        logging.info("calling into the summarization capability")
+        chat_completion_system_context = {
+        "role": "system",
+        "content": [ # this context has to be dynamic according to the request header
+            {
+                "type": "text",
+                # Note: this is a sample summarization prompt which can be tweaked according to your exact needs
+                "text": "You are a useful AI assistant who is an expert at succinctly summarizing long form text into a simple summary. Summarize the text given to you in about 200 words or less."
+            }
+            ]
+        }
+        user_prompt_content = {
+            "type": "text",
+            "text": request_body.get("data", {}).get("text", "")
+        }
+        messages = [
+        chat_completion_system_context,
+        {
+        "role": "user",
+        "content": [user_prompt_content]
+        }
+    ]
+    elif scenario == ENTITY_RECOGNITION_HEADER:
+        logging.info("calling into the entity recognition capability")
+        chat_completion_system_context = {
+        "role": "system",
+        "content": [
+            {
+                    "type": "text",
+                    # Note: this is a sample prompt which can be tweaked according to your exact needs
+                    "text": "You are a useful AI assistant. I need you to help me recognize entities in this piece of text. From the text given to you, identity all people names, addresses, email addresses, engineering job titles and present them as individual lists in a JSON object.",
+                }
+            ]
+        }
+        user_prompt_content = {
+            "type": "text",
+            "text": request_body.get("data", {}).get("text", "")
+        }
+        messages = [
+        chat_completion_system_context,
+        {
+        "role": "user",
+        "content": [user_prompt_content]
+        }
+    ]
+    request_payload = {
+    "messages": messages,
+    "temperature": 0.7,
+    "top_p": 0.95,
+    "max_tokens": 4096

```

Comment:
I think you can also turn on JSON mode to ensure the model's output is to be valid Json.

Reply:
is it something like this? "response_format": { "type": "json_object" }

I'm looking at https://learn.microsoft.com/en-us/azure/ai-studio/reference/reference-model-inference-api?tabs=rest#availability